AppCenter/AppCenter/Internals/HttpClient/MSHttpClient.m

```diff
@@ -2,20 +2,45 @@
 // Licensed under the MIT License.
 
 #import "MSHttpClient.h"
+#import "MSAppCenterInternal.h"
+#import "MSConstants+Internal.h"
+#import "MSHttpCall.h"
+#import "MSHttpClientPrivate.h"
+#import "MSHttpUtil.h"
+#import "MSLoggerInternal.h"
+#import "MSUtility+StringFormatting.h"
+#import "MS_Reachability.h"
 
 @implementation MSHttpClient
 
 - (instancetype)init {
-
-  // TODO implement this properly.
-  return self = [super init];
+  if ((self = [super init])) {
```

Comment:
The initialization pattern here is not really conventional. The initializers in objective c are usually chained as explained in [the Apple documentation](https://developer.apple.com/library/archive/documentation/General/Conceptual/CocoaEncyclopedia/Initialization/Initialization.html) see _Multiple Initializers and the Designated Initializer_.

That means you need to add a designated initializer with max coverage and chain the inits `initWithMaxHttpConnectionsPerHost:retryIntervals:` :
- `init` would call `initWithMaxHttpConnectionsPerHost:retryIntervals:` with default values.
- `initWithMaxHttpConnectionsPerHost:` would call `initWithMaxHttpConnectionsPerHost:retryIntervals:` with default value for retry interval but custom value for max http connexion.
- `initWithRetryIntervals:` would call `initWithMaxHttpConnectionsPerHost:retryIntervals:` with default value for max http connexion but custom value for retry interval.

The main purpose is to enforce the initialization of every properties and not leave one behind when an object is initializing this class or its subclass whatever the initializer used.

Reply:
Oh yeah I had the same comment while pairing and we already fixed it. Should be addressed by 8de53d2